### PR TITLE
Update VersionConverterTask for IndexSpec and allowing Forced updates

### PR DESCRIPTION
**Pull request by drcrallen**
Other PR got swallowed by a rebase. https://github.com/druid-io/druid/pull/1293

In a really basic test with 1 thread vs 1 thread locally, this seems to run about 70% faster than re-indexing the raw data with hadoop.

This does not try to get the most "optimal" way of doing it, but rather simply tries to get a "pretty good" way that has validation.

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/common/task/Task.java [hunk 1]
```diff
@@ -49,8 +49,10 @@
     @JsonSubTypes.Type(name = "index_hadoop", value = HadoopIndexTask.class),
     @JsonSubTypes.Type(name = "index_realtime", value = RealtimeIndexTask.class),
     @JsonSubTypes.Type(name = "noop", value = NoopTask.class),
-    @JsonSubTypes.Type(name = "version_converter", value = VersionConverterTask.class),
-    @JsonSubTypes.Type(name = "version_converter_sub", value = VersionConverterTask.SubTask.class)
+    @JsonSubTypes.Type(name = "version_converter", value = SegmentConverterTask.class), // Backwards compat
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: let's mark those as deprecated while we're at it.

**Comment by nishantmonu51** [comment]
:+1:

File: indexing-service/src/main/java/io/druid/indexing/common/task/Task.java (hunk 1 above)
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: They still exist, just renamed, made a note in code comments

Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: Agree, but I think we should only have one standard way of calling a task, and deprecate others.

**Comment by drcrallen** [comment]
@xvrl  Added simple serde tests

File: indexing-service/src/main/java/io/druid/indexing/common/task/SegmentConverterTask.java [hunk 2]
```diff
@@ -78,37 +98,51 @@ private static String makeId(String dataSource, Interval interval)
...
   )
   {
     super(id, groupId, dataSource, interval);
     this.segment = segment;
     this.indexSpec = indexSpec == null ? new IndexSpec() : indexSpec;
+    this.force = force;
+  }
+
+  @JsonProperty
+  public boolean getForce(){
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/common/task/SegmentConverterTask.java]: isForce

File: indexing-service/src/main/java/io/druid/indexing/common/task/SegmentConverterTask.java [hunk 3]
```diff
@@ -212,6 +251,12 @@ public SubTask(
       );
       this.segment = segment;
       this.indexSpec = indexSpec == null ? new IndexSpec() : indexSpec;
+      this.force = force == null ? false : force;
+    }
+
+    @JsonProperty
+    public boolean getForce(){
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/common/task/SegmentConverterTask.java]: isForce

File: indexing-service/src/test/java/io/druid/indexing/common/task/TaskSerdeTest.java [hunk 4]
```diff
@@ -43,17 +44,25 @@
...
 
 import java.io.File;
+import java.io.IOException;
 
 public class TaskSerdeTest
 {
   private static final ObjectMapper jsonMapper = new DefaultObjectMapper();
 
+  @BeforeClass
+  public static void setUpStatic()
```
Comment (xvrl) [inline indexing-service/src/test/java/io/druid/indexing/common/task/TaskSerdeTest.java]: is this necessary?

File: indexing-service/src/main/java/io/druid/indexing/common/task/Task.java [hunk 5]
```diff
@@ -49,8 +49,10 @@
     @JsonSubTypes.Type(name = "index_hadoop", value = HadoopIndexTask.class),
     @JsonSubTypes.Type(name = "index_realtime", value = RealtimeIndexTask.class),
     @JsonSubTypes.Type(name = "noop", value = NoopTask.class),
-    @JsonSubTypes.Type(name = "version_converter", value = VersionConverterTask.class),
-    @JsonSubTypes.Type(name = "version_converter_sub", value = VersionConverterTask.SubTask.class)
+    @JsonSubTypes.Type(name = "version_converter", value = SegmentConverterTask.class), // Backwards compat - Deprecated
+    @JsonSubTypes.Type(name = "version_converter_sub", value = SegmentConverterTask.SubTask.class), // backwards compat - Deprecated
+    @JsonSubTypes.Type(name = "segment_convert", value = SegmentConverterTask.class),
```
Comment (xvrl) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: I wonder if we should be more consistent with our naming and start with a verb here like we do for most of our tasks, e.g. simply `convert`, and rename the task `ConvertSegmentTask` accordingly?

File: indexing-service/src/test/java/io/druid/indexing/common/task/TaskSerdeTest.java (hunk 4 above)
Comment (drcrallen) [inline indexing-service/src/test/java/io/druid/indexing/common/task/TaskSerdeTest.java]: no removing

File: indexing-service/src/main/java/io/druid/indexing/common/task/Task.java (hunk 5 above)
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/common/task/Task.java]: Fixed

**Pull request merged**